contracts/Comptroller/Comptroller.sol

```diff
@@ -1424,6 +1495,18 @@ contract Comptroller is ComptrollerV9Storage, ComptrollerInterfaceG2, Comptrolle
             "Blacklisted"
         );
 
+        for (uint i = 0; i < allMarkets.length; ++i) {
+            address market = address(allMarkets[i]);
+
+            bool noOriginalSpeed = venusBorrowSpeeds[market] == 0;
+            bool invalidSupply = noOriginalSpeed && venusSupplierIndex[market][user] > 0;
+            bool invalidBorrow = noOriginalSpeed && venusBorrowerIndex[market][user] > 0;
+
+            if (invalidSupply || invalidBorrow) {
+                return amount;
+            }
+        }
```

Comment:
Compound Proposal 62 introduced a bug in the Protocol through which user can claim extra governance token. This Bug was fixed in the Proposal 63 and Proposal 65. They introduced these checks in the Proposal 64 and removed them in Proposal 65 after reseting the amount claimed by the users.
Links to the Comptroller Implementation (Compound):
    [Proposal 62 comptroller Implementation](https://etherscan.io/address/0x374ABb8cE19A73f2c4EFAd642bda76c797f19233#code)
    [Proposal 64 comptroller Implementation](https://etherscan.io/address/0x234b619b4f4e405665f7d94f2ce60c24256032b5/advanced#code)
    [Proposal 65 comptroller Implementation](https://etherscan.io/address/0xBafE01ff935C7305907c33BF824352eE5979B526#code)
    
    We would not require these checks because we did not introduce the bug here.

Reply:
Cool, thanks for the research!